geth/jail/cell.go

```diff
@@ -65,9 +67,30 @@ func (c *Cell) Stop() {
 // event queue loop and schedules for immediate execution.
 // Intended to be used by any cell user that want's to run
 // async call, like callback.
-func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) {
+func (c *Cell) CallAsync(fn otto.Value, args ...interface{}) error {
 	task := looptask.NewCallTask(fn, args...)
-	c.lo.Add(task)
-	// TODO(divan): review API of `loop` package, it's contrintuitive
-	go c.lo.Ready(task)
+	errChan := make(chan error, 1)
+	doneChan := make(chan struct{})
+
+	go func() {
+		errChan <- c.lo.Add(task)
+		// TODO(divan): review API of `loop` package, it's contrintuitive
+		errChan <- c.lo.Ready(task)
+		// If both above return with err == nil,
+		// we signal that the for loop can exit without error
+		doneChan <- struct{}{}
+	}()
+
+	for {
+		select {
+		case err := <-errChan:
+			if err != nil {
+				return err
+			}
+		case <-doneChan:
+			return nil
+		case <-time.After(6000 * time.Millisecond):
```

Comment:
For each call we'll create new time.After that will live minimum 6 seconds. Looks like leak too.
May be the better way is to use 
```
timer := time.NewTimer(6000 * time.Millisecond)
defer timer.Stop()
```
outside of cyrcle and `<-timer.C` inside.